services/brig/src/Brig/API/User.hs

```diff
@@ -227,15 +229,21 @@ createUser new@NewUser{..} = do
         unless av $
             throwE $ DuplicateUserKey k
 
-    createTeam uid activating t = do
-        created <- Intra.createTeam uid t
+    createTeam uid activating t tid = do
+        created <- Intra.createTeam uid t tid
         return $ if activating
                     then Just created
                     else Nothing
 
-    handleTeam (Just (NewTeamMember i))  e = (Nothing, ) <$> findTeamInvitation e i
-    handleTeam (Just (NewTeamCreator t)) _ = return (Just t, Nothing)
-    handleTeam Nothing                   _ = return (Nothing, Nothing)
+    handleTeam :: Maybe NewTeamUser 
+               -> Maybe UserKey
+               -> ExceptT CreateUserError AppIO ( Maybe BindingNewTeamUser
+                                                , Maybe (Team.Invitation, Team.InvitationInfo)
+                                                , Maybe TeamId
+                                                )
+    handleTeam (Just (NewTeamMember i))  e = findTeamInvitation e i
```

Comment:
`(Nothing, ) <$> findTeamInvitation e i` and remove the first return type of findTeamInvitation? findTeamInvidation never returns a BindingNewTeamUser, so it's odd to have that in the type signature.

Reply:
Unfortunately that's not as easy as it seems since `(Nothing, ) <$> findTeamInvitation e i` would then return `(Maybe a, (Maybe b, Maybe c))`

So I could either have a little `do` block or leave as-is, any particular preference?

Reply:
I have a preference for a do block, to keep the type signature of `findTeamInvitation` cleaner.

Reply:
👍 https://github.com/wireapp/wire-server/pull/248/commits/f0fae1f45d934b4a3d3a7ccd49866677538e813a